### ISSUE TITLE
Diagnostic regression for struct-like enum variant constructors from private modules when missing fields

### ISSUE THREAD

**Issue by RadicalZephyr** [rust-lang/rust#80363]
### Code

I tried this code:

```rust
mod inner {
    pub enum MyEnum {
        EscapeSequence { sequence: String, bad: bool },
    }
}

fn main() {
    let x = inner::MyEnum::EscapeSequence {
        sequence: String::from(""),
    };
}
```
[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=55a5a5a399e8d3eb16c4816c98ce0d0d)

I expected to see this happen: 

The compiler used to correctly suggest that a field was missing from the enum constructor:

```
error[E0063]: missing field `bad` in initializer of `inner::MyEnum`
 --> src/main.rs:8:13
  |
8 |     let x = inner::MyEnum::EscapeSequence {
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `bad`

```

Instead, this happened:

Now the compiler erroneously reports:

```
error: cannot construct `MyEnum` with struct literal syntax due to inaccessible fields
 --> src/main.rs:8:13
  |
8 |     let x = inner::MyEnum::EscapeSequence {
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

### Version it worked on

It most recently worked on: Rust 1.47

### Version with regression

`rustc --version --verbose`:
```
rustc 1.48.0 (7eac88abb 2020-11-16)
binary: rustc
commit-hash: 7eac88abb2e57e752f3302f02be5f3ce3d7adfb4
commit-date: 2020-11-16
host: x86_64-unknown-linux-gnu
release: 1.48.0
LLVM version: 11.0
```

**Comment by camelid** [comment]
Duplicate of #79593. Will be fixed in 1.50.
